### PR TITLE
Add Nag for broken Resources

### PR DESCRIPTION
**Pull request by TechnicianLP**
As discussed in #4938 this PR implements a Nag screen when broken or missing texters or models are encountered.
~~This screen is enabled even in non-dev environment to increase pressure on modders to fix their resources.~~ This has since been changed to only be enabled in a development-environment when an offending mod is directory-based
Screen looks like this: https://i.imgur.com/8LBvYNP.png and is shown right before the main menu

### CODE REVIEW

**Comment by KnightMiner** [comment]
If you are going to show it outside a development environment, it needs a Forge config to disable. Most modpack end users do not need to know about missing resources. Hard part is allowing it to be disabled might make people just disable it to ignore, but hopefully mod pack devs and people who make their own packs will be enough to report it properly.

**Comment by covers1624** [comment]
Enabling this out of dev is stupid, users have no reason to see this, and could cause false positives with broken / bad resource packs. Enabling this ONLY in dev, like lex wanted in the first place is what should be done.

**Comment by HenryLoenwind** [comment]
Yeah, if you really want this screen, limit it to dev AND limit it to mods that are loaded from a folder and not a jar file. I do not care about broken resources in all those mods we have compatibility code for.

**Comment by TechnicianLP** [comment]
- Lex wanting it only in Dev was about a hardcrash (as far as i can tell)
- Broken Resourcepacks are indeed a thing 
- Go bug the makers of the broken mods about fixing their shit - dont blame me 

The screen now doesn't show if a ResourcePack is loaded
Added "support" for Recipes

The Problem with disabling it is that it can be abused to disable the screen with an empty ResourcePack

**Comment by asiekierka** [comment]
@TechnicianLP If someone really goes to the trouble of adding an empty resourcepack just to disable it, someone will also go to the trouble of adding an event listener which automatically clicks "Continue". Thus, I don't really see the difference.

**Comment by LexManos** [comment]
Dev-Time

File: src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java [hunk 1]
```diff
@@ -426,6 +427,16 @@ public void onInitializationComplete()
...
             logMissingTextureErrors();
+            if (!badTextureDomains.isEmpty() || !badResources.isEmpty())
+            {
+                if ((Boolean) Launch.blackboard.get("fml.deobfuscatedEnvironment")) {
+                    if (badTextureDomains.size() != 0)
+                        badResources.put("textures", badTextureDomains);
+                    showGuiScreen(new GuiResourceError(badResources));
+                }
+                else
+                    FMLLog.log.warn("Skipped ResourceErrorScreen due to deobfusated Environement");
```
Comment (mezz) [inline src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java]: This log can be removed, it's not really informative.

**Review by mezz — CHANGES_REQUESTED** [review]
Mostly looks good now.

**Review by mezz — APPROVED** [review]

**Comment by HenryLoenwind** [comment]
Again, you DO have the ModContainer the bad resources belong to, you can easily filter out those that come from jar'ed mods...

**Comment by mezz** [comment]
This shouldn't be stale, I just haven't gotten to it. 
I'm not really a big fan of nag screens like this but Lex seems to want it.
@LexManos can you review this one?

**Comment by TechnicianLP** [comment]
still waiting for @LexManos

**Comment by TechnicianLP** [comment]
Will this get merged? I know noone likes to get nagged at for something but looking at the amount of errors in modpacks something has to be done ....

**Comment by TechnicianLP** [comment]
Close or Merge (or request a change)

File: src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java [hunk 2]
```diff
@@ -433,6 +434,14 @@ public void onInitializationComplete()
         else
         {
             logMissingTextureErrors();
+            if (!badTextureDomains.isEmpty() || !badResources.isEmpty())
+            {
+                if ((Boolean) Launch.blackboard.get("fml.deobfuscatedEnvironment")) {
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java]: NPEs?

**Comment by tterrag1098** [comment]
You haven't addressed Henry's concern.

With this change, mods that pull in third-party dependencies will be burdened by their resource errors. Sure this may resolve itself over time, but in the meantime developing will be a nightmare. There *needs* to be a check that the mod in question is loaded from source and not a jar.

**Review by tterrag1098 — CHANGES_REQUESTED** [review]
See above

File: src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java (hunk 2 above)
Comment (bs2609) [inline src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java]: `FMLLaunchHandler.isDeobfuscatedEnvironment()`

Comment (TechnicianLP) [reply src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java]: All occurences of this check are written like this - but might as well change it

**Comment by TechnicianLP** [comment]
Everything should be adressed now

**Review by tterrag1098 — APPROVED** [review]

**Comment by tterrag1098** [comment]
What happens when there are too many errors to show on the screen? Should it perhaps be a scrolling panel?

**Comment by TechnicianLP** [comment]
With the current implementation it currently would just draw beyond the bottom of the screen
The Question is whether a scrollpanel is required as the number of erroring mods in most workspaces should not be enough to go beyond that limit.
If a ScrollPanel is to be implemented nontheless - should i look into generalizing the GuiList implementation in GuiMultipleModsErrored for this purpose?

**Comment by LexManos** [comment]
I'm pretty sure there would be some modders who would easily overflow the screen. So having it scroll would be best. So yes a nice generic scrolling error screen would be best.

**Comment by MajorTuvok** [comment]
Any update on this? In my opinion such a Nag screen is a nice and in some cases needed thing... (which is why it shouldn't be stale)

**Comment by TechnicianLP** [comment]
As this PR got delayed and 1.13 having disposed of GuiScrollingList (likely a relict from beta1.7.3 as the mcp43 names suggest) in favor of the vanilla implementation (GuiSlot) - this PR is now using that 1.13 route for scrolling screens as well

**Comment by MajorTuvok** [comment]
This should not go stale in my opinion...

**Comment by LexManos** [comment]
This needs updating to 1.14, 1.12 wont be getting this.

**Pull request closed without merging**